### PR TITLE
Test examples to see, if everything is working with TRAVIS

### PR DESCRIPTION
**Pull request by ue71603**
A first set of test examples.

### CODE REVIEW

**Comment by ue71603** [comment]
New version.

**Comment by ue71603** [comment]
I left currently only few request /response. This should show travis is working. However, I want to know if the prefixes and the style are ok.  I will now add some examples for other services too.

**Review by sgrossberndt — CHANGES_REQUESTED** [review]
`examples/test_examples_switzerland/TR_001.xml` and `examples/test_examples_switzerland/TR_001.xml.tmp` are empty files, I suppose `examples/test_examples_switzerland/TR_001.xml` should contain the matching request to `examples/test_examples_switzerland/TR_001_response.xml `?

Maybe use speaking names for the files too: `LIR_002_trip_from_st_gallen_to_luzern_request.xml` / `LIR_002_trip_from_st_gallen_to_luzern_response.xml`?

File: .travis/xmllint-check.sh [hunk 1]
```diff
@@ -2,3 +2,5 @@
 /usr/bin/find . -type f \( -name "*.xsd" -or -name "*.xml" \) -print0 | while read -r -d $'\0' filename; do XMLLINT_INDENT=$'\t' xmllint --encode UTF-8 --pretty 1 "${filename}" > "${filename}.pretty"; mv "${filename}.pretty" "${filename}"; grep -i -v "xmlspy" "${filename}" > "${filename}.tmp" && mv "${filename}.tmp" "${filename}"; done;
 echo "finished formatting"
 # xmllint --noout --schema OJP.xsd examples/subdirectory1/*xml examples/subdirectory2/*xml
+xmllint --noout --schema OJP.xsd examples/test_examples_switzerland/*xml
```
[the remark this answers is not in the document]
Comment (ue71603) [reply .travis/xmllint-check.sh:5]: changed

File: .travis/xmllint-check.sh [hunk 2]
```diff
@@ -2,3 +2,5 @@
 /usr/bin/find . -type f \( -name "*.xsd" -or -name "*.xml" \) -print0 | while read -r -d $'\0' filename; do XMLLINT_INDENT=$'\t' xmllint --encode UTF-8 --pretty 1 "${filename}" > "${filename}.pretty"; mv "${filename}.pretty" "${filename}"; grep -i -v "xmlspy" "${filename}" > "${filename}.tmp" && mv "${filename}.tmp" "${filename}"; done;
 echo "finished formatting"
 # xmllint --noout --schema OJP.xsd examples/subdirectory1/*xml examples/subdirectory2/*xml
```
[the remark this answers is not in the document]
Comment (ue71603) [reply .travis/xmllint-check.sh:4]: changed

File: examples/test_examples_switzerland/LIR_001_search_stop_with_name_Zuerich_Kreuzplatz.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (ue71603) [reply examples/test_examples_switzerland/LIR_001_search_stop_with_name_Zuerich_Kreuzplatz.xml:1]: changed

**Comment by sgrossberndt** [comment]
We have a build failure, which is not caused by the examples but by the OJP schema definition:
https://travis-ci.com/github/VDVde/OJP/builds/189204061#L134
@pvgrumbkow @skinkie @herlitze

**Comment by pvgrumbkow** [comment]
> We have a build failure, which is not caused by the examples but by the OJP schema definition:
> https://travis-ci.com/github/VDVde/OJP/builds/189204061#L134
> @pvgrumbkow @skinkie @herlitze
>
Apparently multiple imports of SIRI XSDs cause the problem, right? But what would be the solution?

**Comment by skinkie** [comment]
@pvgrumbkow the rationale here was: inform the SIRI WG. My previous attempt: fix this locally.

**Comment by pvgrumbkow** [comment]
Hmm, the warning message doesn't sound like a SIRI issue to me. It states that OJP_Requests.xsd tries to import a SIRI XSD which was already imported by OJP.xsd. Where is the SIRI issue then?
By the way: This is a warning. Why does it prevent building?

**Comment by ue71603** [comment]
I think the problem is with the way the import is done in the examples. I will try some other things.  I wasn't really happy there, but it worked wid XML Spy.
I hope that I can do this this evening, so we have examples tomorrow.

**Comment by ue71603** [comment]
sgrossberdt: Why is your change still blocking? As far as I can see, it should be resolved.

**Comment by sgrossberndt** [comment]
As I wrote the issues which make the build fail are not due to your examples, @ue71603 - the build failure is caused by the OJP XSD schema definition itself.

**Comment by sgrossberndt** [comment]
Ok, so I identified several issues:

The ominous *.tmp files which are occurring stem from Travis CI and have not committed by @ue71603 - this happens when the linting/prettifying from https://github.com/VDVde/OJP/blob/changes_for_v1.1/.travis/xmllint-check.sh#L2 fails and leaves a *.tmp file - which is then found as difference by Travis and committed, as visible in https://github.com/VDVde/OJP/pull/85/commits/ba7ea1b4e91ff98b203be1a44c99c9d7a3514ece?file-filters%5B%5D=

The Travis log contains warnings and errors, the warnings due to the namespace "siri" being imported multiple times with different files in it, e.g. `OJP.xsd` defines SIRI as its main namespace in `xs:schema`: `xmlns="http://www.siri.org.uk/siri"`, defines the location for this schema as `<xs:include schemaLocation="siri/siri_base-v2.0.xsd"/>` and imports the namespace `http://www.vdv.de/ojp` with schema location `OJP_Requests.xsd`.
But `OJP_Requests.xsd` defines the SIRI namespace again and does this by referencing *another* SIRI XSD file: `<xs:import namespace="http://www.siri.org.uk/siri" schemaLocation="./siri/siri_requests-v2.0.xsd"/>` and also includes `OJP_Fare.xsd` which once again defines the SIRI namespace by referencing *another* SIRI XSD file: `<xs:import namespace="http://www.siri.org.uk/siri" schemaLocation="./siri_model/siri_modes-v1.1.xsd"/>`

This apparently leads to issues as some parsers (xmllint, Xerces) cannot cope with the same namespace name being bound to different schema files in imported files. Also read https://stackoverflow.com/questions/4998063/one-xml-namespace-equals-one-and-only-one-schema-file

My proposal to fix the SIRI includes in the XSD is to always use `./siri_model/siri_all-v2.0.xsd` instead of trying to shorten the number of includes SIRI files at the expense of an invalid XSD.
This will fix the SIRI includes but the examples will still be invalid, as we have the same issue with the namespace `xmlns="http://www.vdv.de/ojp"` too. Here I am still investigating on how to fix it.

**Comment by sgrossberndt** [comment]
I propose to drop this branch and close the pull request in favor of https://github.com/VDVde/OJP/pull/115

**Comment by sgrossberndt** [comment]
Closed in favor of https://github.com/VDVde/OJP/pull/115

**Pull request closed without merging**